test/range_test.py

```diff
@@ -455,7 +458,78 @@ def test_start_long_before_long_hours_back_and_with_long_hours_forward(self):
+
+    def _test_filesystems_and_globs(self, datetime_to_task, datetime_to_re, expected):
+        actual = list(_get_filesystems_and_globs(datetime_to_task, datetime_to_re))
+        self.assertEqual(len(actual), len(expected))
+        for (actual_filesystem, actual_glob), (expected_filesystem, expected_glob) in zip(actual, expected):
+            self.assertTrue(isinstance(actual_filesystem, expected_filesystem))
+            self.assertEqual(actual_glob, expected_glob)
+
+    def test_date_glob_successfully_inferred(self):
+        self._test_filesystems_and_globs(
+            lambda d: CommonDateTask(d),
+            lambda d: '(%04d).*(%02d).*(%02d)' % (d.year, d.month, d.day),
+            [
+                (MockFileSystem, '/n2000y01a05n/[0-9][0-9][0-9][0-9]_[0-9][0-9]-_-[0-9][0-9]aww/21mm01dara21'),
+            ]
+        )
+
+    def test_datehour_glob_successfully_inferred(self):
+        self._test_filesystems_and_globs(
+            lambda d: CommonDateHourTask(d),
+            lambda d: '(%04d).*(%02d).*(%02d).*(%02d)' % (d.year, d.month, d.day, d.hour),
+            [
+                (MockFileSystem, '/n2000y01a05n/[0-9][0-9][0-9][0-9]_[0-9][0-9]-_-[0-9][0-9]aww/21mm[0-9][0-9]dara21'),
+            ]
+        )
+
+    def test_wrapped_datehour_globs_successfully_inferred(self):
+        self._test_filesystems_and_globs(
+            lambda d: CommonWrapperTask(d),
+            lambda d: '(%04d).*(%02d).*(%02d).*(%02d)' % (d.year, d.month, d.day, d.hour),
+            [
+                (MockFileSystem, 'TaskA/[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]'),
+                (MockFileSystem, 'TaskB/no/worries[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]'),
+            ]
+        )
+
+    def test_inconsistent_output_datehour_glob_not_inferred(self):
+        class InconsistentlyOutputtingDateHourTask(luigi.Task):
+            dh = luigi.DateHourParameter()
+
+            def output(self):
+                base = self.dh.strftime('/even/%Y%m%d%H')
+                if self.dh.hour % 2 == 0:
+                    return MockFile(base)
+                else:
+                    return {
+                        'spi': MockFile(base + '/something.spi'),
+                        'spl': MockFile(base + '/something.spl'),
+                    }
+
+        with self.assertRaises(NotImplementedError):
+            list(_get_filesystems_and_globs(
+                lambda d: InconsistentlyOutputtingDateHourTask(d),
+                lambda d: '(%04d).*(%02d).*(%02d).*(%02d)' % (d.year, d.month, d.day, d.hour)))
+
+    def test_wrapped_inconsistent_datehour_globs_not_inferred(self):
+        class InconsistentlyParameterizedWrapperTask(luigi.WrapperTask):
+            dh = luigi.DateHourParameter()
+
+            def requires(self):
+                yield TaskA(dh=self.dh - datetime.timedelta(days=1))
+                yield TaskB(dh=self.dh, complicator='no/worries')
+
+        with self.assertRaises(NotImplementedError):
```

Comment:
Not under 2.6 https://travis-ci.org/spotify/luigi/jobs/51823860#L775

I will add fix in #777